src/hoc/withPreventNavigation/modalWithPreventNavigation.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import * as _ from 'underscore';
+
+import {closeModal, IModalActionPayload} from '../../components/modal/ModalActions';
+import {IReduxAction, ReduxConnect} from '../../utils/ReduxUtils';
+import {IWithDirtyProps, withDirty} from '../withDirty/withDirty';
+import {PreventNavigationPrompt} from './PreventNavigationPrompt';
+
+export interface IWithPreventNavigationConfig {
+    id: string;
+    title?: string;
+    content?: React.ReactNode;
+    exit?: string;
+    stay?: string;
+}
+
+export interface IWithPreventNavigationInjectedProps extends IWithDirtyProps {
+    validateShouldNavigate: (isDirty: boolean) => boolean;
+}
+
+export interface IWithPreventNavigationDispatchProps {
+    closeModal: (id: string) => void;
+}
+
+export interface IWithPreventNavigationState {
+    showPrevent: boolean;
+}
+
+export const preventNavigationDefaultConfig: Partial<IWithPreventNavigationConfig> = {
+    title: 'Unsaved Changes',
+    content: 'Are you sure you want to leave this page without saving? All unsaved changes will be lost.',
+    exit: 'Exit without applying changes',
+    stay: 'Cancel',
+};
+
+export const modalWithPreventNavigation = <T, R = any>(config: IWithPreventNavigationConfig) => (Component: React.ComponentClass<T, R>): React.ComponentClass<T & Partial<IWithPreventNavigationInjectedProps>, R> => {
+    const mapDispatchToProps = (dispatch: (action: IReduxAction<IModalActionPayload>) => void): IWithPreventNavigationDispatchProps => ({
+        closeModal: (id: string) => dispatch(closeModal(id)),
+    });
+
+    @ReduxConnect(undefined, mapDispatchToProps)

```

Comment:
Same here, you could define mapDispatchToProps as an object instead
```suggestion
    @ReduxConnect(null, {closeModal})
```